### PR TITLE
Implement warning for ignored trailing arguments

### PR DESCRIPTION
**Pull request by In-line**

### CODE REVIEW

**Comment by In-line** [comment]
https://stackoverflow.com/questions/67903849/whats-the-different-between-cargo-build-and-cargo-build

File: src/bin/cargo/cli.rs [hunk 1]
```diff
@@ -169,6 +170,17 @@ fn expand_aliases(
                     cmd,
                 ))?;
             }
+            (Some(_), None) => {
+                // Command is built-in and is not conflicting with alias, but contains ignored values.
+                if let Some(mut values) = args.values_of("") {
+                    config.shell().warn(format!(
```
Comment (In-line) [inline src/bin/cargo/cli.rs:176]: should this be error?

Comment (ehuss) [reply src/bin/cargo/cli.rs:176]: I think an error would be fine.

Comment (In-line) [reply src/bin/cargo/cli.rs:176]: @ehuss It can break backward compatability

Comment (ehuss) [reply src/bin/cargo/cli.rs:176]: If you're concerned about back compat, that's fine too. It can always be changed to an error in the future if desired.

**Comment by ehuss** [comment]
Thanks!

@bors r+

**Pull request merged**